astropy/utils/data.py

```diff
@@ -232,6 +233,7 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         try:
             import bz2
         except ImportError:
+            fileobj.close()
```

Comment:
Yeah -- I suppose the finally does handle it.  Sorry about that -- reading the diffs I missed the larger context.